lib/faraday/retry/middleware.rb

```diff
@@ -124,6 +129,15 @@ def retry_statuses
       #   the the value of the retry header and should return the number of
       #   seconds to wait before retrying the request. This is useful if the
       #   value of the header is not a number of seconds or a RFC 2822 formatted date.
+      # @option options [Block] :exhausted_retries_block block will receive
+      #   when all attempts are exhausted. The block will be yielded keyword arguments:
+      #     * any args [Hash]: Anything you need to solve the problem.
+      #       The transferred hash will be available in the block with a simple call.
+      #     * env [Faraday::Env]: Request environment
+      #     * exception [Exception]: exception that triggered the retry,
+      #       will be the synthetic `Faraday::RetriableResponse` if the
+      #       retry was triggered by something other than an exception.
+      #     * options [Faraday::Options]: middleware options
```

Comment:
My suggestion: the documentation here should not speak of how you have configured the argument list of the exhausted retries block proc - if you extend its argument list to also finesse in some extra information at run-time, that's not what the block gets called with.

```suggestion
      # @option options [Block] :exhausted_retries_block block will receive
      #   when all attempts are exhausted. The block will be yielded keyword arguments:
      #     * env [Faraday::Env]: Request environment
      #     * exception [Exception]: exception that triggered the retry,
      #       will be the synthetic `Faraday::RetriableResponse` if the
      #       retry was triggered by something other than an exception.
      #     * options [Faraday::Options]: middleware options
```

Telling people about the "you can add extra proc arguments  besides the three which are required, to meet your application-specific needs" is _good_. But I think that this is not where they should be told about it.

Reply:
Ok, I understood... :thinking: 
Thanks, Olle, for your explanation!

Sorry for the mistakes in the doc.. =(

I wil fix it